src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,8 +50,12 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+            for (String caseInsensitiveNames : wordsInName){
+                for (String caseInsensitiveKeyword : keywords){
+                    if (caseInsensitiveNames.equalsIgnoreCase(caseInsensitiveKeyword)){
```

Comment:
Would using `disjoint` have a better performance than this O(N^2) solution? 

Also, thought this solution might be a little convoluted due to the 4 layers of nesting.

Reply:
Using `Collections.disjoint()` would be easier to read (: